ethkey/src/secret.rs

```diff
@@ -61,51 +66,91 @@ impl Secret {
 
 	/// Inplace add one secret key to another (scalar + scalar)
 	pub fn add(&mut self, other: &Secret) -> Result<(), Error> {
-		let mut key_secret = self.to_secp256k1_secret()?;
-		let other_secret = other.to_secp256k1_secret()?;
-		key_secret.add_assign(&SECP256K1, &other_secret)?;
+		match (self.is_zero(), other.is_zero()) {
```

Comment:
The reason behind these changes is that starting with this PR, I need to handle 0 EC scalars.
0 is the invalid key in secp256k1 && library will return an error when trying to work with it.
Long story short: it wasn't a best idea to use `Secret` for EC arithmetic (the main assumption was that it will minimize `U256` <-> `Secret` convertations).
And the general solution is either to find&use/create separate EC-arithmetic library.
If you're strong against these changes - please comment && I'll close this PR && will switch to separate type for EC calculations first.

Reply:
not strongly against these changes, but down the road it could be nice to have a separate type for secp256k1 scalars which has `std::ops` traits implemented for it.